File: Talleres-2B/Taller-4/Funcion4-2B.py
print("---Función para verificar si un número es impar y primo---")#Método principal

#Ingreso de datos
num = int(input("Ingrese el número a verificar: "))

#Inicialización de las banderas
esImpar = True and False
esPrimo = True and False

#Función Primo
def Primo (numero):
    div = 0

    for i in range (numero):
        if (numero % i == 0):
            div = div + 1
    
    if (div == 2):
        return True

    else: 
        return False

#Función Impar
def Impar(numero):
    if (numero % 2 == 0):
        return False
    else:
        return True

#Asignarle un valor a la función
esImpar = Impar(num)

#Proceso
if (esImpar == True):
    esPrimo = Primo(num)

    if (esPrimo == True):
        print("El número ",num," es impar y es primo")
    
    else:
        print("El número ",num," es impar pero no es primo")

else:
    print("El número ",num," no es impar, por lo tanto no se verifica que sea primo")

File: Talleres-2B/Taller-4/FuncionesYProced/src/funcionesyproced/Funcion4_2B.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package funcionesyproced;
import java.util.Scanner;
import static funcionesyproced.Funcion3_2B.Impar;
/**
 *
 * @author Santiago Espinoza
 */
public class Funcion4_2B {
    public static void main(String[] args) {
        Scanner leer = new Scanner(System.in);
        
        int num = 0;
        
        boolean esImpar, esPrimo;
        
        System.out.println("\tFunción para verificar si un número es impar y primo");
        
        System.out.print("\nIngrese el número a verificar: ");
        num = leer.nextInt();
        
        esImpar = Impar(num);
        if (esImpar == true){
            esPrimo = Primo(num);
            if (esPrimo == true){
                System.out.println("\nEl número " +num +" es impar y es primo");
            }
            else{
                System.out.println("\nEl número " +num + " es impar y no es primo");
            }
        }
        else {
            System.out.println("El número no es impar por lo tanto no se verifica que sea primo");
        }
    }
    public static boolean Primo(int numero) {
        int div = 0;
        
        for (int i = 1; i<=numero; i++){
            if (numero % i == 0){
                div = div + 1;
            }
        }
        if (div == 2){
            return true;
        }
        else{
            return false;
        }
    }
}
